File: Data Structures & Algorithms/car-fleet/submission-3.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        vector<pair<int, int>> pos_sp;
        int n = position.size(); 
        for(int i = 0; i < n; i++){
            pos_sp.push_back({position[i], speed[i]});
        }

        sort(pos_sp.begin(), pos_sp.end());

        stack<double> res; 
        for(auto elem: pos_sp){
            int pos = elem.first; 
            int sp = elem.second; 
            int miles = target - pos; 
            double time = (double)miles / sp; 
            while(!res.empty() && res.top() <= time){
                res.pop();
            }
            res.push(time);
        }
        return res.size();
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = [] 
        curr = [] 

        def dfs(i: int, sm: int) -> None: 
            if i >= len(nums): 
                return 

            if sm == target: 
                res.append(curr.copy()) 
                return 
            
            if sm > target: 
                return 
            
            curr.append(nums[i])
            dfs(i, sm + nums[i]) 
            curr.pop() 
            dfs(i+1, sm) 

        dfs(0, 0) 
        return res 